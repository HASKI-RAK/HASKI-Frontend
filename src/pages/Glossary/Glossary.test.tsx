import { Glossary } from "@pages"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import { AuthContext } from "@services"
import { MemoryRouter } from "react-router-dom"
import { render } from "@testing-library/react"

const navigate = jest.fn()

describe("Test the Glossary page", () => {
    global.fetch = jest.fn(() =>
        Promise.resolve({
            json: () => Promise.resolve({ status: 200 }),
            status: 200,
            message: "OK",
        }),
    ) as jest.Mock

    it("Should render the skeleton", () => {
        const glossary = render(
            <MemoryRouter>
                <AuthContext.Provider value={{ isAuth: false, setIsAuth: jest.fn(), logout: jest.fn() }}>
                    <Glossary />
                </AuthContext.Provider>
            </MemoryRouter>
        )
        expect(glossary.container.querySelectorAll("span").length).toEqual(1);
    })

    it("Should render the glossary page", () => {
        const glossary = render(
            <MemoryRouter>
                <AuthContext.Provider value={{ isAuth: true, setIsAuth: jest.fn(), logout: jest.fn() }}>
                    <Glossary />
                </AuthContext.Provider>
            </MemoryRouter>
        )
        expect(glossary.container.querySelectorAll("span").length).toEqual(1);
    })
    
    test("renders correctly and matches snapshot", () => {
        const tree = renderer.create(<Glossary />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})