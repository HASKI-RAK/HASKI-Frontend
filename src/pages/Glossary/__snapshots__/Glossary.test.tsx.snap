// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Glossary page renders correctly and matches snapshot 1`] = `
[
  <div>
    <h3
      className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Glossar
    </h3>
  </div>,
  <div>
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
      />
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="MuiFormControl-root css-2m9kme-MuiFormControl-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            id="glossary-filter-label"
          >
            Tags
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              disabled={false}
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Tags
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="outlined-basic"
              id="outlined-basic-label"
            >
              Suche
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="outlined-basic"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Suche
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </h4>
      </div>
    </div>
    <h4
      className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      GlossaryIndex
    </h4>
  </div>,
  <div>
    <div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              className="MuiBox-root css-qxh7fn"
            >
              <div
                className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                Adaptierbarkeit
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    Die Adaptierbarkeit eines Systems bedeutet, dass dieses durch externe Aktionen hinsichtlich seiner Programmeinstellungen verändert werden kann. Das HASKI-System ist für Lehrpersonen und für die Lernenden adaptierbar. Während Lehrpersonen z.B. in die vorgeschlagenen Lernpfade eingreifen können, können die Lernenden selbstständig bestimmte Grundeinstellungen modifizieren. Letzteres kann z.B. das Verändern von Farben oder die Konfigurierung einer eigenen Sequenz von Lernelementen sein (vgl. Zens 2005; Petko & Reusser 2005, S.188; Bartel, Dörringer & Klopp 2022a).
                  </div>
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        Quellen:
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
                      >
                        Bartel, Dörringer & Klopp, 2022a; Petko & Reusser, 2005, S.188; Zens, 2005
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              className="MuiBox-root css-qxh7fn"
            >
              <div
                className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                Adaptivität
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    Adaptivität ist in einem System vorhanden, wenn es sich selbstständig an den Input (z.B. Testergebnisse, Bearbeitungsgeschwindigkeit, Lernwege usw.) anpasst. „Adaptivität ermöglicht in kurzen Abständen eine kontinuierliche Anpassung des Systems“ (Petko & Reusser 2005, S.188). In diesem Kontext meint adaptives Lernen „Anpassung des Lernangebots an die individuellen Unterschiede der Lernenden, damit alle gleichermaßen gefördert werden können“ (Stangl 2022, S.1) (vgl. Bartel, Dörringer & Klopp 2022a).
                  </div>
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        Quellen:
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
                      >
                        <span
                          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                          style={
                            {
                              "height": undefined,
                              "width": "40%",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              className="MuiBox-root css-qxh7fn"
            >
              <div
                className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                Adaptive Inhaltspräsentation
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": "80%",
                        }
                      }
                    />
                  </div>
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        Quellen:
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
                      >
                        <span
                          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                          style={
                            {
                              "height": undefined,
                              "width": "40%",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1elwnq4-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded={false}
          className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-sh22l5-MuiButtonBase-root-MuiAccordionSummary-root"
          id="panel1a-header"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
          >
            <div
              className="MuiBox-root css-qxh7fn"
            >
              <div
                className="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                <span
                  className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                  style={
                    {
                      "height": undefined,
                      "width": undefined,
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style={
            {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="panel1a-header"
                className="MuiAccordion-region"
                role="region"
              >
                <div
                  className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                    <span
                      className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                      style={
                        {
                          "height": undefined,
                          "width": "80%",
                        }
                      }
                    />
                  </div>
                  <div
                    className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                      >
                        Quellen:
                      </div>
                      <div
                        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 css-ykovpn-MuiGrid-root"
                      >
                        <span
                          className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave css-1etr2k8-MuiSkeleton-root"
                          style={
                            {
                              "height": undefined,
                              "width": "40%",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
